docs/gallery/general/plot_resources.py

```diff
@@ -0,0 +1,396 @@
+er = HERD()
+file = HERDManagerContainer(name='file')
+
+data = Data(
+    name='data_name',
+    data=np.array(
+        [('Mus musculus', 9, 81.0), ('Homo sapiens', 3, 27.0)],
+        dtype=[('species', 'U14'), ('age', 'i4'), ('weight', 'f4')]
+    )
+)
+
+er.add_ref(
+    file=file,
+    container=data,
+    field='species',
+    key='Mus musculus',
+    entity_id='NCBI_TAXON:txid10090',
+    entity_uri='https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=10090'
+)
+
+###############################################################################
+# Write HERD
+# ------------------------------------------------------
+# :py:class:`~pynwb.resources.HERD` is written as a zip file of
+# the individual tables written to tsv.
+# The user provides the path, which contains the name of the file.
+
+er.to_zip(path='./HERD.zip')
+
+###############################################################################
+# Read HERD
+# ------------------------------------------------------
+# Users can read :py:class:`~pynwb.resources.HERD` from the zip file
+# by providing the path to the file itself.
+
+er_read = HERD.from_zip(path='./HERD.zip')
+os.remove('./HERD.zip')
+
+##################################################
+# Steaming an entire Dandiset for HERD
+# ---------------------------------
+#
+# A single :py:class:`~pynwb.resources.HERD` instance can contain references for
+# multiple :py:class:`~pynwb.file.NWBFile` objects. We support both ffspec and ROS3
+# streaming to remotely access files from the DANDI Archive without having to download
+# the memory intensive set of files themselves. With streaming, users can easily annotate
+# datasets and attributes within existing :py:class:`~pynwb.file.NWBFile` objects.
+# Retrieve urls for all nwbfiles in the dandiset
+dandiset_id = '000015'
+filepath = 'sub-anm372795/sub-anm372795_ses-20170718.nwb'  # 450 kB file
+with DandiAPIClient() as client:
+    asset = client.get_dandiset(dandiset_id, 'draft')
+    urls = []
+    for asset_id in tqdm(asset.get_assets()):
+        path = asset_id.path
+        file_asset = asset.get_asset_by_path(path)
+        s3_url = file_asset.get_content_url(follow_redirects=1, strip_query=True)
+        urls.append(s3_url)
+# TODO: Add a quick description of the Dandiset
+
+# Create HERD instance
+herd= HERD()
+
+# first, create a virtual filesystem based on the http protocol
+fs = fsspec.filesystem("http")
+
+# create a cache to save downloaded data to disk (optional)
+fs = CachingFileSystem(
+    fs=fs,
+    cache_storage="nwb-cache",  # Local folder for the cache
+)
+
+# Iteratively populate HERD
+for url in tqdm(urls):
+    with fs.open(url, "rb") as f:
+        with h5py.File(f) as file:
+            with pynwb.NWBHDF5IO(file=file, load_namespaces=True) as io:
+                read_file = io.read()
+                # ADD HERD for Subject species
+                entity = herd.get_entity(entity_id='NCBI_TAXON:10090')
```

Comment:
If we are already fetching the entity, then it would be nice if `add_ref` accepted the entity as input, rather than having to specify `entity_id` and `entity_uri` each time.